ff/serving/jobs/materialize_job_test.go

```diff
@@ -0,0 +1,205 @@
+package jobs
+
+import (
+	"errors"
+	"fmt"
+	"golang.org/x/sync/errgroup"
+	"math"
+	"testing"
+)
+
+type MockMaterializedFeatures struct {
+	Rows []FeatureRow
+}
+
+type JobTestParams struct {
+	FeatureTableData []interface{}
+	NumJobs          int
+}
+
+type TestError struct {
+	Outcome string
+	Err     error
+}
+
+func (m *TestError) Error() string {
+	return fmt.Sprintf("%v: %s", m.Err, m.Outcome)
+}
+
+func (m *MockMaterializedFeatures) NumRows() (int, error) {
+	return len(m.Rows), nil
+}
+
+func (m *MockMaterializedFeatures) IterateSegment(begin int, end int) (FeatureIterator, error) {
+	return &MockFeatureIterator{
+		CurrentIndex: 0,
+		Slice:        m.Rows[begin:end],
+	}, nil
+}
+
+type MockOnlineTable struct {
+	DataTable map[string]interface{}
+}
+
+func (m *MockOnlineTable) Set(entity string, value interface{}) error {
+	m.DataTable[entity] = value
+	return nil
+}
+
+func (m *MockOnlineTable) Get(entity string) (interface{}, error) {
+	value, exists := m.DataTable[entity]
+	if !exists {
+		return nil, errors.New("Value does not exist in online table")
+	}
+	return value, nil
+}
+
+type MockFeatureIterator struct {
+	CurrentIndex int
+	Slice        []FeatureRow
+}
+
+func (m *MockFeatureIterator) Next() bool {
+	m.CurrentIndex++
+	return m.CurrentIndex < len(m.Slice)
+
+}
+
+func (m *MockFeatureIterator) Err() error {
+	return nil
+}
+
+func (m *MockFeatureIterator) Entity() string {
+	return m.Slice[m.CurrentIndex].Entity
+}
+
+func (m *MockFeatureIterator) Value() interface{} {
+	return m.Slice[m.CurrentIndex].Row
+}
+
+type FeatureRow struct {
+	Entity string
+	Row    interface{}
+}
+
+func testFollowingParams(params *JobTestParams) error {
+
+	featureRows := make([]FeatureRow, len(params.FeatureTableData))
+	for i, row := range params.FeatureTableData {
+		featureRows[i] = FeatureRow{Entity: fmt.Sprintf("entity_%d", i), Row: row}
+	}
+
+	materialized := &MockMaterializedFeatures{Rows: featureRows}
+
+	table := &MockOnlineTable{
+		DataTable: make(map[string]interface{}),
+	}
+
+	jobList := make([]*MaterializedChunkRunner, params.NumJobs)
```

Comment:
No need for all of this, just test that it copys a chunk from materialized to online table